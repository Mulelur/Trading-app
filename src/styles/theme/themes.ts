const lightTheme = {
  primary: '#12161E',
  text: '#12161E',
  textSecondary: 'rgb(102, 191, 191)',
  background: 'rgb(234, 246, 246)',
  backgroundVariant: 'rgb(102, 191, 191)',
  border: 'rgb(102, 191, 191)',
  borderLight: 'rgb(196, 212, 237, .3)',
};

const darkTheme: Theme = {
  primary: 'rgba(220,120,95,1)',
  text: 'rgba(241,233,231,1)',
  textSecondary: 'rgba(241,233,231,0.6)',
  background: 'rgba(0,0,0,1)',
  backgroundVariant: 'rgba(28,26,26,1)',
  border: 'rgba(241,233,231,0.15)',
  borderLight: '#2c2525',
};

export type Theme = typeof lightTheme;

export const themes = {
  light: lightTheme,
  dark: darkTheme,
};
