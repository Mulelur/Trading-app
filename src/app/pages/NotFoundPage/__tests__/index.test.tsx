import * as React from 'react';
import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styles/theme/ThemeProvider';
import { HelmetProvider } from 'react-helmet-async';
import { configureAppStore } from 'store/configureStore';
import { NotFoundPage } from '..';
import { Route, Routes, BrowserRouter } from 'react-router-dom';

const renderNotFoundPage = (store: Store) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <BrowserRouter>
          <HelmetProvider>
            <Routes>
              <Route path="/" element={<NotFoundPage />} />
            </Routes>
          </HelmetProvider>
        </BrowserRouter>
      </ThemeProvider>
    </Provider>,
  );

describe('<NotFoundPage />', () => {
  let component: ReturnType<typeof renderNotFoundPage>;

  beforeEach(() => {
    const store = configureAppStore();
    component = renderNotFoundPage(store);
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });
});
