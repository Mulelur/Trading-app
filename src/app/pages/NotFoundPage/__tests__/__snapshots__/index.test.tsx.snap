// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match snapshot 1`] = `
.c10 {
  font-size: 0.8rem;
  line-height: 1.5;
  color: rgba(58,52,51,0.7);
  margin: 0.225rem 0 1.5rem 0;
}

.c11 {
  color: rgba(215,113,88,1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c11:active {
  opacity: 0.4;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
  color: rgba(215,113,88,1);
}

.c7 {
  color: rgba(215,113,88,1);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.15rem 0.3rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:hover {
  opacity: 0.8;
}

.c7:active {
  opacity: 0.4;
}

.c7 .icon {
  margin-right: 0.2rem;
}

.c1 {
  display: none;
}

.c2 {
  color: rgba(215,113,88,1);
  padding-right: 0.8rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-size: 1.25rem;
  color: rgba(58,52,51,1);
  font-weight: bold;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  padding-right: 0.5rem;
}

.c0 {
  box-shadow: 0 1px 0 0 rgba(58,52,51,0.05);
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: rgba(255,255,255,1);
  z-index: 2;
  width: 100%;
  padding-left: 10rem;
  padding-right: 10rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  height: calc(100vh - 4rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 320px;
}

.c9 {
  margin-top: -8vh;
  font-weight: bold;
  color: rgba(58,52,51,1);
  font-size: 3.375rem;
}

.c9 span {
  font-size: 3.125rem;
}

@media only screen and (max-width:768px) {
  .c1 {
    display: block;
  }
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {
  .c0 {
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: rgba(255,255,255,0.75);
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    padding-left: 1rem;
    padding-right: 3rem;
  }
}

@media only screen and (min-width:600px) {
  .c0 {
    padding-left: 5rem;
    padding-right: 5rem;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    padding-left: 17rem;
    padding-right: 7rem;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    padding-left: 18rem;
    padding-right: 8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class=""
      >
        <div
          class="c2"
        >
          <svg>
            menu-icon.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/"
      >
        <div
          class="c5"
        >
          <svg>
            logo-icon.svg
          </svg>
        </div>
        Weblinnk
      </a>
    </div>
    <div
      class="c6"
    >
      <a
        class="c7"
        href="/profile"
        rel="noopener noreferrer"
        target="_blank"
        title="Documentation Page"
      >
        <svg>
          account-icon.svg
        </svg>
      </a>
      <a
        class="c7"
        href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
        rel="noopener noreferrer"
        target="_blank"
        title="Github Page"
      >
        <svg>
          bell-icon.svg
        </svg>
      </a>
    </div>
  </div>
  <div
    class="c8"
  >
    <div
      class="c9"
    >
      4
      <span
        aria-label="Crying Face"
        role="img"
      >
        ðŸ˜¢
      </span>
      4
    </div>
    <p
      class="c10"
    >
      Page not found.
    </p>
    <a
      class="c11"
      href="/"
    >
      Return to Home Page
    </a>
  </div>
</div>
`;
